packages/react-reconciler/src/ReactFiberHydrationContext.js

```diff
@@ -577,6 +672,23 @@ function prepareToHydrateHostSuspenseInstance(fiber: Fiber): void {
   hydrateSuspenseInstance(suspenseInstance, fiber);
 }
 
+function skipPastDehydratedActivityInstance(
+  fiber: Fiber,
+): null | HydratableInstance {
+  const activityState: null | ActivityState = fiber.memoizedState;
+  const activityInstance: null | ActivityInstance =
+    activityState !== null ? activityState.dehydrated : null;
+
+  if (!activityInstance) {
```

Comment:
>_possible bug_

**✨ AI Suggestion:** The variable `activityInstance` is already checked for nullity in the outer `if` statement. The nested null check `!activityInstance` is redundant.
```suggestion
    const activityState: null | ActivityState = fiber.memoizedState;
    const activityInstance: null | ActivityInstance =
      activityState !== null ? activityState.dehydrated : null;

    if (activityInstance === null) {
```